target/linux/ath79/dts/ar9331_onion_omega.dts

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: GPL-2.0
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar9331.dtsi"
+
+/ {
+	model = "Onion Omega";
+	compatible = "onion,omega", "qca,ar9331";
+
+	aliases {
+		serial0 = &uart;
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		expansion-dock-blue {
+			label = "exp-dock-rgb:blue:expansion-dock-blue";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		expansion-dock-green {
+			label = "exp-dock-rgb:green:expansion-dock-green";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+		expansion-dock-red {
+			label = "exp-dock-rgb:red:expansion-dock-red";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+
+		led_system: system {
+			label = "onion:amber:system";
+			gpios = <&gpio 27 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		button0 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&ref {
+	clock-frequency = <25000000>;
+};
+
+&uart {
+	status = "okay";
+};
+
+&gpio {
+	status = "okay";
+};
+
+&usb {
+	status = "okay";
+
+	dr_mode = "host";
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&eth0 {
+	status = "okay";
+
+	compatible = "syscon", "simple-mfd";
+};
+
+&eth1 {
+	status = "okay";
+
+	mtd-mac-address = <&uboot 0x01fc00>;
+	mac-address-increment = <(-1)>;
```

Comment:
mtd mac will _work_ but openwrt master just recently moved to the "nvmem" style.  This _should_ be fairly straightforward to figure out from the recent commits...

Reply:
Thanks got it resolved, I think. Compiles and runs both with a sysupgrade and a factory flash. Thanks!